### PR TITLE
Fix handling of dictionary-style sounds for macOS 10.15

### PR DESCRIPTION
**Pull request by zacwest**
This syntax doesn't make sense for non-critical alerts, but we should not fail badly at it.

Fixes home-assistant/iOS#1229.

### CODE REVIEW

File: functions/legacy.js [hunk 1]
```diff
@@ -81,7 +81,16 @@ module.exports = {
 
           if((typeof req.body.registration_info.os_version === "string")
             && (req.body.registration_info.os_version.startsWith('10.15'))) {
-            payload.apns.payload.aps.sound = path.parse(payload.apns.payload.aps.sound).name
+            switch(typeof payload.apns.payload.aps.sound) {
+              case "string":
+                payload.apns.payload.aps.sound = path.parse(payload.apns.payload.aps.sound).name
```
Comment (JBassett) [inline functions/legacy.js:86]:
```suggestion
                payload.apns.payload.aps.sound = path.parse(payload.apns.payload.aps.sound).name;
```

File: functions/legacy.js [hunk 2]
```diff
@@ -81,7 +81,16 @@ module.exports = {
...
+                break;
+              case "object":
+                if(typeof payload.apns.payload.aps.sound.name === "string") {
+                  payload.apns.payload.aps.sound.name = path.parse(payload.apns.payload.aps.sound.name).name
```
Comment (JBassett) [inline functions/legacy.js:90]:
```suggestion
                  payload.apns.payload.aps.sound.name = path.parse(payload.apns.payload.aps.sound.name).name;
```

**Review by JBassett — CHANGES_REQUESTED** [review]
Just some `;` for consistency with the rest of the file.

**Comment by zacwest** [comment]
Thanks! JavaScript is very JavaScript!

**Review by JBassett — APPROVED** [review]

**Pull request merged**